ext/NormalizingFlowsEnzymeCoreExt.jl

```diff
@@ -0,0 +1,39 @@
+module NormalizingFlowsEnzymeCoreExt
+
+using EnzymeCore
+using NormalizingFlows
+using NormalizingFlows: ADTypes, DifferentiationInterface
+
+# deal with Enzyme readonly error: see https://discourse.julialang.org/t/enzyme-autodiff-readonly-error-and-working-with-batches-of-data/123012
+function NormalizingFlows._prepare_gradient(loss, adbackend::ADTypes.AutoEnzyme, θ, args...)
+    if isempty(args)
+        return DifferentiationInterface.prepare_gradient(
+            EnzymeCore.Const(loss), adbackend, θ
+        )
+    end
+    return DifferentiationInterface.prepare_gradient(
+        EnzymeCore.Const(loss),
+        adbackend,
+        θ,
+        map(DifferentiationInterface.Constant, args)...,
+    )
+end
+
+function NormalizingFlows._value_and_gradient(
+    loss, prep, adbackend::ADTypes.AutoEnzyme, θ, args...
+)
+    if isempty(args)
+        return DifferentiationInterface.value_and_gradient(
+            EnzymeCore.Const(loss), prep, adbackend, θ
+        )
+    end
+    return DifferentiationInterface.value_and_gradient(
+        EnzymeCore.Const(loss),
+        prep,
+        adbackend,
+        θ,
+        map(DifferentiationInterface.Constant, args)...,
```

Comment:
What's the difference between `EnzymeCore.Const` and `DI.Constant`?

Reply:
`DI.Constant` translates into `Enzyme.Const` for the Enzyme backend, and to other things for other backends. It is used to signal that the corresponding argument is not differentiated